### PR TITLE
eng, bug fix, Verify TypeSpec Code Generation step should fail, if codegen report error

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Description

There is a case that emitter error, but CI passes.
https://github.com/Azure/azure-sdk-for-java/pull/38535

https://dev.azure.com/azure-sdk/public/_build/results?buildId=3514795&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=d31d5e2d-d8df-53bc-da94-87e19f5b2ee4

We should fail at this Verify TypeSpec Code Generation.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
The failure on "java - azure-communication-messages" is what this PR plan to do.

```
Generate code failed for following modules:
/mnt/vss/_work/1/s/sdk/communication/azure-communication-messages
```

**Review by haolingdong-msft — APPROVED** [review]

File: eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1 [hunk 1]
```diff
@@ -36,15 +36,27 @@ Invoking tsp-client update
...
+    $failedSdk += $sdkPath
+  }
   Pop-Location
 }
 
+if ($failedSdk.Length -gt 0) {
+  Write-Host "
+Generate code failed for following modules:
+$failedSdk
+"
```
Comment (srnagar) [inline eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1:56]: nit: move these lines into a single line for easy readability
```suggestion
  Write-Host "Code generation failed for following modules: $failedSdk"
```

**Review by srnagar — APPROVED** [review]

**Pull request merged**